### DISCUSSION TITLE
Slow starting cypress tests

### DISCUSSION THREAD

**Discussion by RobbeCl** [cypress-io/cypress#26901] [Questions and Help]
When running Cypress locally, everything runs blazingly fast, but in CI/CD the startup time takes around 4 minutes (consistently between chrome/firefox/electron headless). 

We are using the docker image [cypress/browsers:node18.12.0-chrome107](https://hub.docker.com/layers/cypress/browsers/node18.12.0-chrome107/images/sha256-50959d1166a52276536e5535d1671692b83c097921685fe6a17f020d0b0f6ccf?context=explore). We are also using a strong AWS machine (t3a.2xlarge) so processing should be a problem I think. Besides that we are using an [esbuild preprocessor](https://github.com/bahmutov/cypress-esbuild-preprocessor). 

Some logging info:
It hangs around 4 minutes and then prints "Run Starting" (with all the browser infromation + test suites). After it launches the browser (after 4 minutes) everyting runs fast and smooth. It just the startup-time which is slow. Behaviour is the same on e2e tests as component tests.

Could it be the esbuild preprocessing step which takes a long tome or is it something browser-related?

**Comment by RobbeCl** [comment]
@lmiller1990 do you have an idea what the problem could be? Thanks! :-)

**Reply by lmiller1990** [reply]
That sounds very unusual. Are you using Vite or webpack for your dev server? The preprocessor is only used for End to End testing.

What is your project like / how many spec files? Do you find the same issue when you reduce the specs files to a smaller number (eg - is the issue related to the specs / dev server, or a more deep seeded issue?)

**Reply by lmiller1990** [reply]
Any chance of a minimal reproduction? I'll move this to an issue.

**Reply by RobbeCl** [reply]
I removed all our tests but one, and I used the Included cypress Docker container to work with a prebuild binary. It still takes a long time to start the first test (aka launch the browser?). Then I added a specPattern to the component tests, but still no improvement. 

It keeps hanging for a few minutes on this step:
```
> cypress run --component --browser chrome
[STARTED] Task without title.
[SUCCESS] Task without title.
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
DevTools listening on ws://127.0.0.1:42021/devtools/browser/39551398-4597-4817-bd7b-0eee97ea2add
[575:0707/130859.302378:ERROR:gpu_memory_buffer_support_x11.cc(44)] dri3 extension not supported.
(node:597) ExperimentalWarning: Custom ESM Loaders is an experimental feature and might change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
(node:597) ExperimentalWarning: The Node.js specifier resolution flag is experimental. It could change or be removed at any time.
```

This is our config:
```ts
import { defineConfig } from 'cypress';
import { cypressEsbuildPreprocessor } from 'cypress-esbuild-preprocessor';

export default defineConfig({
  projectId: 'project-name',
  viewportWidth: 1900,
  viewportHeight: 1060,
  video: false,
  e2e: {
    defaultCommandTimeout: 6000,
    baseUrl: 'http://localhost:5173',
    setupNodeEvents(on) {
      on(
        'file:preprocessor',
        cypressEsbuildPreprocessor({
          esbuildOptions: {
            sourcemap: false,
          },
        })
      );
    },
  },
  chromeWebSecurity: false,

  component: {
    specPattern: 'src/**/*.cy.{js,jsx,ts,tsx}',
    defaultCommandTimeout: 6000,
    devServer: {
      framework: 'vue',
      bundler: 'vite',
    },
  },
});

```

Its not really possible to make a reproduction, because it is working locally like a charm.

**Reply by RobbeCl** [reply]
After some more research, we saw the same slowdown happening if we used the Docker container locally on my colleague his pc. However, my pc did not show such a slowdown.

**Reply by lmiller1990** [reply]
So it is slow but only inside docker? Are you able to provide a repo (since it's a docker related issue, by the sounds of things, it sounds like providing a repo should be more practical now)?
